src/gennylib/test/PoolFactory_test.cpp

```diff
@@ -231,15 +231,15 @@ TEST_CASE("PoolFactory behavior") {
     }
 
     SECTION("PoolManager can construct multiple pools") {
-        auto yaml = YAML::Load(R"(
-        SchemaVersion: 2018-07-01
-        Database: test
-        Actors: []
-        )");
-        genny::ActorHelper ah{yaml, 1, {}};
-        auto w = ah.workload();
-        w->client("Foo", 0);
-        w->client("Foo", 10);
-        w->client("Bar", 1);
+        genny::v1::PoolManager manager{"mongodb:://localhost:27017", {}};
+        genny::v1::ConfigNode config{YAML::Load(R"()")};
+
+        auto foo0 = manager.client("Foo", 0, config);
+        auto foo0again = manager.client("Foo", 0, config);
+        auto foo100 = manager.client("Foo", 10, config);
+        auto bar0 = manager.client("Bar", 0, config);
+
+        REQUIRE((manager.instanceCount() ==
+                 std::unordered_map<std::string, size_t>({{"Foo", 2}, {"Bar", 1}})));
```

Comment:
[note] This assertion doesn't actually verify that we aren't calling `createPool()` again when running `manager.client("Foo", 0, config)` a second time. A different style of trying to write this test is to register a callback which gets called by `createPool()` and use that to "spy on" the `name` and `instance` for which `createPool()` gets called. I'm fine leaving this as-is but just wanted to mention it as another example of where something like [TIG-1191](https://jira.mongodb.org/browse/TIG-1191) would probably be helpful.

Reply:
Ya I considered going to that extreme but didn't feel worth it. I added a comment to this test (borrowing most of your words) for future maintainers. If this is ever seriously refactored that's a consideration that should be taken into mind.